Milestone 2: Implement basic game flow and UI

cyonii opened the pull request:
Implementation of the user flow and the user interface

- Add Player class
- Add Game class
- Add Game #methods

cyonii committed:
Implement basic game flow and UI

layzrd reviewed (changes requested):
### Status: Require changes  ♲
Hi @Misori-simon  and @cyonii , 👏 You did very well on this project. ✨ Clap 👏👏👏 for accepting players' names. Most requirements have done 👌. Kindly fix the following issues.

[REQUIRED]

- [x]  The game should inform players if the selected move is invalid (without computing a move)
- [x]  The game should display the board after the player move (without computing which spots on the board are taken)
- [x]  Fix the blank lines that present on the console after accepting the player's name

Once you fix the issues, kindly submit another code review. ♲

Happy coding 👨‍💻
 [Abdulaziz Redi Ali](https://azizali.ml/)

Misori-simon committed:
add puts for invalid cell number

cyonii committed:
Implement user flow of Tic Tac Toe game

- we implemented a mock flow of how 2 players are expected to interact 
with a Tic Tac Toe game

ElijahTobs reviewed (changes requested):
# Status: Changes Required :recycle: 

## Hi @cyonii and @Misori-simon 

You two have done a good job so far and well done implementing the changes requested by the previous TSE. :+1: 

However, there are still some changes you need to effect before this milestone can be approved.

## [REQUIRED]
- [ ] Game informs player if the selected move is a draw move (without computing draw)


## Work on the required change and submit for a new code review

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

Misori-simon committed:
add game tie clause
-add a game tie clause to display game tie;  if the game is a tie

ElijahTobs reviewed (approved):
# Status: Approved! :tada: 

Congratulations! @Misori-simon and @cyonii 

### Good job implementing the required change

### You can now merge and continue with the next milestone.

Happy Coding :computer: 

Feel free to reach out to me 
slack @ElijahTobs
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)
